### PR TITLE
Catch StopIteration exception when no endpoint left alive

### PR DESCRIPTION
**Pull request by JordanP**

### CODE REVIEW

**Comment by scality** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://37.187.159.67:5443/job/scality-sproxyd-client-tox/24/
Test PASSed.

File: scality_sproxyd_client/sproxyd_client.py [hunk 1]
```diff
@@ -142,6 +142,13 @@ def __repr__(self):
         return ret % (self.sproxyd_urls_set, self.conn_timeout,
                       self.proxy_timeout)
 
+    def get_url_for_object(self, name):
```
Comment (NicolasT) [inline scality_sproxyd_client/sproxyd_client.py]: Docstring

Comment (JordanP) [inline scality_sproxyd_client/sproxyd_client.py]: Done

**Comment by scality** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://37.187.159.67:5443/job/scality-sproxyd-client-tox/25/
Test PASSed.

File: scality_sproxyd_client/sproxyd_client.py [hunk 2]
```diff
@@ -142,6 +142,19 @@ def __repr__(self):
...
+        '''Get an absolute URL from which the object `name` can be accessed.
+
+        Only healthy Sproxyd endpoints are concidered to form the base object
+        path. The object `name` is properly escaped by `urllib.quote` which
+        preserves '/' by default.
+        '''
+        try:
+            sproxyd_url = self.sproxyd_urls.next()
+        except StopIteration:
+            raise exceptions.SproxydException("No Sproxyd endpoint alive !")
```
Comment (NicolasT) [inline scality_sproxyd_client/sproxyd_client.py]: Not a real blocker (guess I missed this one before), but: there's no need to add punctuation (or certainly no exclamation marks) to exception messages :wink:

**Comment by NicolasT** [comment]
LGTM

File: scality_sproxyd_client/sproxyd_client.py (hunk 2 above)
Comment (JordanP) [inline scality_sproxyd_client/sproxyd_client.py]: I guess I wanted to add some drame.

I should have gone with "OOhhh nnnooessss, something went wrong !!!!!!!!!!1"

**Pull request merged**